docs/products/clickhouse/concepts/indexing.rst

```diff
@@ -1,43 +1,58 @@
+This article helps you understand how ClickHouse® processes data and how it differs from the way the other database management systems (DBMS) do that. Discover spare and skipping indexes and check out what the vector computation engine is used for.
 
-ClickHouse is a columnar database. This means that ClickHouse can read columns selectively retrieving only necessary information and omitting columns that are not needed for the request. The smaller the number of the columns you read, the faster and more efficient the performance of the request. If you have to read many or all columns, then using a columnar database becomes a less effective approach.
+ClickHouse as a columnar database
+---------------------------------
 
-.. note:: Read more about columnar databases :doc:`in our separate article <columnar-databases>`.
+ClickHouse is a columnar database, which means that ClickHouse can read columns selectively retrieving only necessary information and omitting columns that are not needed for the request. The smaller the number of the columns you read, the faster and more efficient the performance of the request. If you have to read many or all columns, using a columnar database becomes a less effective approach.
 
-ClickHouse is designed to process substantial chunks of information for a single request. To make this possible it shifts the focus from reading individual lines into scanning massive blocks of data for a request. This happens by using what is called **a vector computation engine**, which reads data in blocks that normally consist of thousands of rows, selected for a small set of columns.
+.. seealso::
+    
+    Read more about characteristics of columnar databases and their features in :doc:`Columnar databases <columnar-databases>`.
+
+Reading data in blocks
+----------------------
+
+ClickHouse is designed to process substantial chunks of information for a single request. To make this possible, it shifts the focus from reading individual lines into scanning massive blocks of data for a request. This happens by using what is called the vector computation engine, which reads data in blocks that normally consist of thousands of rows selected for a small set of columns.
 
 ClickHouse primary index
 ------------------------
 
-To find a specific block, ClickHouse uses **a primary index**. However, unlike in other systems the primary index in ClickHouse has a couple of distinctive characteristics.
+ClickHouse reads data in blocks. To find a specific block, ClickHouse uses its primary index, which has a few distinctive features that makes it different from primary indexes of the other systems.
+
+Instead of indexing every row, ClickHouse indexes very 10000th row (or, to be precise, every 8192nd row when using default settings). Such type of index is descriptively called *sparse index* and the batch of rows is sometimes called *granule*.
```

Comment:
```suggestion
Instead of indexing every row, ClickHouse indexes every 10000th row (or, to be precise, every 8192nd row when using default settings). Such index type is descriptively called *sparse index*, and the batch of rows is sometimes called *granule*.
```

Reply:
Good you're flagging this one - an article is also missing :-)